### PR TITLE
satellite/admin-back-office-v1: Create doc for initial release

### PR DESCRIPTION
**Pull request by ifraixedes**
Create the design doc for the new satellite administration back office.

This kicks start the review iteration process.

Closes https://github.com/storj/team-delivery/issues/289

### CODE REVIEW

File: satellite/admin-back-office-v1/user-roles.drawio [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<mxfile host="Electron" modified="2023-08-09T11:37:20.511Z" agent="Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/21.6.5 Chrome/114.0.5735.243 Electron/25.3.1 Safari/537.36" etag="m65WNdth2jxJov2d9NVG" version="21.6.5" type="device">
```
Comment (egonelbre) [inline satellite/admin-back-office-v1/user-roles.drawio:1]: Tip, you should be able to use mermaid diagrams. The main benefit is that modifying the diagram inline in the markdown is more likely to stay up to date.

In this case you could create a subgraph per entity (e.g. Account, Project, Bucket) and then have nodes for each role and draw a line to them. Alternatively a C4 diagram could work.

File: satellite/admin-back-office-v1.md [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+allows to interact with its endpoints through Web forms.
+
+API authentication is conducted by two mechanism:
+- [An unique token secret](https://github.com/storj/storj/blob/main/satellite/admin/server.go#L215).
+  The service isn't exposed publicly, the only way to access to it is forwarding the port of the
+  corresponding Kubernetes pod.
+- [Allowed groups](https://github.com/storj/storj/blob/main/satellite/admin/server.go#L231).
+  This requires that the server is accessed through an authorization proxy (e.g.
+  [Oauth2 proxy](https://oauth2-proxy.github.io/)):
+  - AP1: https://ap1.admin.storj.tools
```
Comment (egonelbre) [inline satellite/admin-back-office-v1.md:37]: Doesn't it make more sense to keep this on `storj.io`, because it's critical infrastructure?

File: satellite/admin-back-office-v1.md [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+  5. Remove data placement restriction (geo fence).
+3. Bucket
+  1. View information: ID, name, user agent, data placement restriction (geo fence).
+  2. Set user agent.
+  3. Set data placement restriction (geo fence).
+  4. Remove data placement restriction (geo fence).
+
+##### Job stories
+
+__When a user wants to access__
```
Comment (egonelbre) [inline satellite/admin-back-office-v1.md:104]: Lets use `######` here. If there's some github bug that doesn't allow h6, then we can remove one layer by either naming these as `Story: XYZ` or making the `Operations` and such a level higher.

File: satellite/admin-back-office-v1.md [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+
+__When a user wants to change the email address tied to their account__
+
+I want to be able to edit the email address in the account details view so that all events and data
+associated to that user is now under the updated email address.
+
+##### Audit trail
+
+The back office keeps a register of all performed operations and who executed them that cause
+modifications.
```
Comment (egonelbre) [inline satellite/admin-back-office-v1.md:297]: We probably should keep track of `reading` data as well, not just modification.

File: satellite/admin-back-office-v1.md [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+##### API implementation
+
+We have to review the current satellite administration API to see what endpoints we need to add or
+modify.
+
+We also have to adapt it to allow to perform the operations based on the usr roles specified in the
+_Authentication / Authorization_ section and be able to configure the groups associated to them.
+
+We have to review it to see if we want to add more logging or metrics, considering that wasn't
+created as a main feature as now.
```
Comment (egonelbre) [inline satellite/admin-back-office-v1.md:470]: Note, I would prefer this to be generated both for the Go api side and typescript side.

File: satellite/admin-back-office-v1.md [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+### Security
+
+Everything related to uses' authentication relies on the external authentication proxy. A shallow
+list of proxy responsibilities are:
+- Users account management.
+- Users' sessions management.
+- Security on the authentication method.
+
+Because of the operations that the back office allows, we have to pay special attention and take all
+the measures that we can for protecting the site from Cross-Site Request Forgery (a.k.a. CSRF).
```
Comment (egonelbre) [inline satellite/admin-back-office-v1.md:494]: There's also https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html

File: satellite/admin-back-office-v1.md [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+We write integration tests for each API endpoint, including successful, error responses due to
+invalid requests, unauthenticated, and authorized requests.
+
+We coordinate with our QA team to establish a test plan for the Web UI.
+
+### Rollout
+
+We plan to use the same rollout procedure that we are having with the current admin UI/API.
+
+## Rollback
```
Comment (egonelbre) [inline satellite/admin-back-office-v1.md:534]: I didn't notice any mentioning of how and when it will be deployed. Similarly, with zero-downtime upgrades we might have to think about automatically refreshing the frontend when an upgrade happens.

File: satellite/admin-back-office-v1.md [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+_Authentication / Authorization_ section and be able to configure the groups associated to them.
+
+We have to review it to see if we want to add more logging or metrics, considering that wasn't
+created as a main feature as now.
+
+##### Web UI
+
+We have to build a new Web UI from scratch with the Vue framework and the Vuetify component
+framework. The Web UI is already in development and it almost have every component and page that the
+back office requires, you can see it on https://storj.github.io/admin-ui/login.
```
Comment (egonelbre) [inline satellite/admin-back-office-v1.md:476]: This seems to show a lot more information that necessary.

File: satellite/admin-back-office-v1.md (hunk 2 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:37]: Perhaps, but this is detailing the current status and they are served under storj.tools domain

File: satellite/admin-back-office-v1.md (hunk 3 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:104]: I didn't use 6 because of Github incompatibilities, I wanted to have that at the same level as _Operations_.

File: satellite/admin-back-office-v1.md (hunk 4 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:297]: That wasn't specified in the requirements document https://docs.google.com/document/d/1KzZuhfvyU1CgrQm9L4moYxtQME_V6zTYuc6pdarKrUk/edit#heading=h.a7bdzb7w5g2a

What is it going to provide? This audit trail is shown on certain pages (e.g. see at the bottom of https://storj.github.io/admin-ui/account-details)

I understand that we may want to log/audit the user access to information, but we could keep that in a log rather than in the DB

Comment (egonelbre) [reply satellite/admin-back-office-v1.md:297]: Ah, yes, that makes sense. I just want to make sure that reading information also needs some sort of audit trail.

File: satellite/admin-back-office-v1.md (hunk 5 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:470]: I'd love that option.
Can you confirm to me if that's an implementation requirement or if it has to be discussed with others, in that case, who?

File: satellite/admin-back-office-v1.md (hunk 8 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:476]: I mentioned data aggregation that we don't have for example at the account level.
I'm not sure if we have that information for buckets either.

@stoweandi indicated to @boshevski to remove it (there is a Slack thread discussion where I mentioned), however I don't know if @boshevski had the time to do it.

File: satellite/admin-back-office-v1.md (hunk 5 above)
Comment (egonelbre) [reply satellite/admin-back-office-v1.md:470]: It's not a requirement. We have https://github.com/storj/storj/tree/main/private/apigen that we should be able to use -- although, might need some modifications or improvements depending on the exact needs.

You can ask @mobyvb or @jewharton for the latest status of that package.

File: satellite/admin-back-office-v1.md (hunk 6 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:494]: I mentioned those cheatsheets series below (last paragraph of this section).
However, I didn't find an index page, so I linked the first cheatsheet of the series. 
I'm going to update the link

File: satellite/admin-back-office-v1/user-roles.drawio (hunk 1 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1/user-roles.drawio:1]: I would have loved to draw a mermaid one, but this kind of diagram cannot be done and I believe that this kind is clearer than one with boxes and arrows.

File: satellite/admin-back-office-v1.md (hunk 4 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:297]: I will rename it.
Do you think that we should also have the audit trail in a log format in this version or we should write it in the _out of scope_ section?

File: satellite/admin-back-office-v1.md (hunk 2 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:37]: Do you think that we should clarify in this document which domain should serve the new back office?

Comment (egonelbre) [reply satellite/admin-back-office-v1.md:37]: If this is a design doc, then yes, if this documents what the current behavior is, then no.

File: satellite/admin-back-office-v1.md (hunk 4 above)
Comment (egonelbre) [reply satellite/admin-back-office-v1.md:297]: Audit trail should definitely part of the initial version.

File: satellite/admin-back-office-v1.md (hunk 7 above)
Comment (egonelbre) [reply satellite/admin-back-office-v1.md:534]: > What do you mean?

When either user of the admin panel has it open, we should automatically refresh the browser, when there's an upgrade to the backend API. There's a potential of sending an old API request to a new server.

Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:534]: In the end, the API shouldn't allow performing an operation that we have withdrawn in a new release.
I cannot imagine how an old frontend can make an API call to the new release that has changed and do what the user doesn't expect.

The only thing that I see is that the frontend shows errors.

Isn't that a common issue for any web application?

The only way that I think how to mitigate is the frontend sends a kind of version or random value that we rotate on each release and the API rejects any request that doesn't have that value and the frontend auto-refresh when it receives a specific error because of that.

File: satellite/admin-back-office-v1.md (hunk 5 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:470]: Ok, I will update the document with that.

File: satellite/admin-back-office-v1.md (hunk 4 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:297]: I'll add that audit trail to log to a file using a specific logger only for it.

File: satellite/admin-back-office-v1.md (hunk 2 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:37]: this document is a design doc for the new back office, this section is the context that describes the current system.

File: satellite/admin-back-office-v1/user-roles.drawio (hunk 1 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1/user-roles.drawio:1]: That's cool, I will add markdown table.

File: satellite/admin-back-office-v1.md (hunk 7 above)
Comment (egonelbre) [reply satellite/admin-back-office-v1.md:534]: > I cannot imagine how an old frontend can make an API call to the new release that has changed and do what the user doesn't expect.

Let's say you add a new field to json that allows `null` as a value. Then the browser sending the request would add a "null" field, even though the UI might've wanted to preserve the previous value.

> Isn't that a common issue for any web application?

Kind of yes.

> The only way that I think how to mitigate is the frontend sends a kind of version or random value that we rotate on each release and the API rejects any request that doesn't have that value and the frontend auto-refresh when it receives a specific error because of that.

You can store the "expected api version" (e.g. commit hash, version tag, binary hash) in the UI and then use that (assuming that the admin server will serve the UI). If not, then the UI could request it during startup.

Comment (egonelbre) [reply satellite/admin-back-office-v1.md:534]: (Note, it's not critical for the first version)

Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:534]: > You can store the "expected api version" (e.g. commit hash, version tag, binary hash) in the UI and then use that (assuming that the admin server will serve the UI). If not, then the UI could request it during startup.

I said random, but could be any of those.
We'll need a server to serve the UI, it could be the same API server or not. If it isn't we could have a configuration field for setting it.

> (Note, it's not critical for the first version)

I'll add it the _out of scope_ section.

File: satellite/admin-back-office-v1.md (hunk 2 above)
Comment (egonelbre) [reply satellite/admin-back-office-v1.md:37]: Then I would probably mention it in some other section.

File: satellite/admin-back-office-v1.md [hunk 9]
```diff
@@ -0,0 +1,617 @@
...
+
+We need to add the API calls and UI logic (client side validations, operation confirmation, etc.)
+and all the Javascript client side code that those require.
+
+We use the Javascript API client generated by
```
Comment (mobyvb) [inline satellite/admin-back-office-v1.md:517]: here are some additional links of actual implementation of the generator which could be helpful. Of course, we will want a separate implementation for the back office, but these could be a good reference:
* [API definition](https://github.com/storj/storj/blob/main/satellite/console/consoleweb/consoleapi/gen/main.go). In theory, this is the only one you need to modify manually. Everything else is generated at the paths defined in this doc.
* [Generated backend code](https://github.com/storj/storj/blob/main/satellite/console/consoleweb/consoleapi/api.gen.go) based on definition
* [Generated documentation](https://github.com/storj/storj/blob/main/satellite/console/consoleweb/consoleapi/apidocs.gen.md) based on definition
* [Generated typescript/frontend code](https://github.com/storj/storj/blob/main/web/satellite/src/api/v0.gen.ts) based on definition.

File: satellite/admin-back-office-v1.md [hunk 10]
```diff
@@ -0,0 +1,617 @@
...
+##### Operations
+
+1. Account:
+  1. View information: ID, email, full name, paid/free tier, suspended?, user agents, Multi-factor
+     authentication enabled/disabled, data placement restriction (geo fence), list of projects,
+     limits (download, storage, number of segments, number of projects allowed).
+  2. Change email address.
+  3. Disable Multi-factor authentication.
+  5. Set limits (download, storage, number of segments, number of projects allowed).
+  6. Suspend "disable upload/download".
```
Comment (mobyvb) [inline satellite/admin-back-office-v1.md:80]: For the purposes of this document, I think this terminology is fine. However, I want to emphasize that we will need to develop some standard universal terminology to distinguish different types of "suspension" for accounts. We currently have multiple ways for a user to become "suspended", and there is different expected behavior for each:
* "billing suspension/freeze" - when a user does not pay invoices, disable upload/download
  * should be unfrozen automatically as soon as invoices are paid
* "prod owner suspension/freeze" - when a user uploads/shares illegal or malicious content, in violation of our terms of service
  * should be permanent, with no way to automatically exit

Right now, we have a lot of confusion about these two situations and there is not clear shared vocabulary across teams, e.g. when Console says "freeze" we are usually referring to "billing freeze", but when support or prod owner says "freeze", they may be talking about a "permanent suspension"

The main takeaway for this back office doc is that we should not go into this thinking that there is only one type of suspend/disable/freeze/whatever we call it - and we should make the differences between these obvious to the operator using the back office.

For the record, I don't have a preference for what terms to use for these different types. But I think we should pick a single word like "freeze" or "suspend" or "disable", and combine it with a second word indicating the type, e.g. "billing freeze" or "permanent freeze" :shrug:

File: satellite/admin-back-office-v1.md [hunk 11]
```diff
@@ -0,0 +1,617 @@
...
+  because it could be more than one modified field per operation.
+- Previous data. The existing data previous to the modification performed by this operation. This is
+  a structured blob because it could be more than one modified field per operation.
+
+This information is stored in the same database that the satellite uses.
+
+The _DBX_ schema is
+
+```
+model admin_operation_modifications_history (
```
Comment (mobyvb) [inline satellite/admin-back-office-v1.md:317]: even if we do not use it at first, I think it would be a good idea to include a "notes" or "metadata" field in this table

That way, in atypical situations, the operator would have the option of adding additional notes about what they did/why. I wouldn't expect to use it in most cases, but I imagine it could be nice in unusual circumstances to be able to add optional meta notes about the operation

**Review by mobyvb — APPROVED** [review]
the main idea lgtm, but I added some additional comments that you may want to look at before merging

File: satellite/admin-back-office-v1.md (hunk 10 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:80]: Are those different types already available? if yes, please link where I can see them.

> For the record, I don't have a strong preference for what terms to use for these different types. But I think we should pick a single word like "freeze" or "suspend" or "disable", and combine it with a second word indicating the type, e.g. "billing freeze" or "permanent freeze" 🤷

I'd rather use one of those single words with a second word indicating the type too. If I have to vote for one of those words, I vote for "suspend".

File: satellite/admin-back-office-v1.md (hunk 11 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:317]: At this time the back office doesn't have the option to add notes to any of the operations that offers.

Adding a column to the table that can be empty and not reference any other entity is an easy task that can be done through a migration; it will require a lot more work to add that notes/metadata field to every API endpoint and modify the UI that the database change, so I feel that adding the column at this stage is a "premature feature" (a term adopted from "premature optimization" :joy:) and it may confuse people how read the code in the future.

File: satellite/admin-back-office-v1.md (hunk 9 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md]: I add those references to the document.

File: satellite/admin-back-office-v1.md (hunk 10 above)
Comment (mobyvb) [reply satellite/admin-back-office-v1.md:80]: The types for the "account freezes" table at the time of this comment are [here](https://github.com/storj/storj/blob/c706df52185d99f651963b2e4538b8937310ead3/satellite/console/accountfreezes.go#L70-L78)
Currently, they are just called `Freeze` and `Warning`, and both of them are explicitly related to unpaid invoices (billing suspend).
In order to "permanent suspend" right now, the prod owner must use "set limits" on the user and set to 0, and cannot use "freeze", or the user can exit.
However, soon, I would like to rename `Freeze` and `Warning` to `BillingFreeze` and `BillingWarning`, and add a third type in the section of code I linked to function as "permanent suspend"

TLDR: the type for "permanent suspend" is not available yet, but I think we should add it so that the prod owner does not need to use "set limits" in order to do this

File: satellite/admin-back-office-v1.md (hunk 11 above)
Comment (mobyvb) [reply satellite/admin-back-office-v1.md:317]: ok good point, and I agree.

File: satellite/admin-back-office-v1.md (hunk 10 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:80]: Ok, I'll add that new type of suspension

Comment (ifraixedes) [reply satellite/admin-back-office-v1.md]: Done.
@mobyvb please review that change and the roles that have permissions to perform each one. The changes are in a separated commit, so it should be easy for you to just only check that commit.

File: satellite/admin-back-office-v1.md
Comment (dlamarmorgan) [inline satellite/admin-back-office-v1.md:1]: I don't see any mention of node tagging,  which was recently added to the satellite to manage not only things like country placement (geo fence), but also things like projects that require SOC2 compliance. Is it possible to generalize the mention of  geo fence to this more generic "view of node tag requirements" for a project/bucket?

File: satellite/admin-back-office-v1.md [hunk 12]
```diff
@@ -0,0 +1,636 @@
...
+
+I want authorized users to see a simple "reactivate" button in the user's detail view.
+
+Acceptance criteria:
+- Button resumes a user's limits to what they had them set at before suspension occurred.
+- Add an optional admin note of the reason for unsuspending.
+
+__When an account needs a limit increase or decrease__
+
+I want authorized users to be able to input the allotted usage for:
```
[the remark this answers is not in the document]
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:252]: I'm not sure, this was copied from Google docs. The current admin API doesn't have a rate limit change.

image


@stoweandi what do you think?
On the other hand, the current admin API doesn't even have an account limit for a number of buckets, can you tell me if that has to be kept or removed?

File: satellite/admin-back-office-v1.md [hunk 13]
```diff
@@ -0,0 +1,636 @@
...
+structure when the performed operations is an entity update and when the operations is:
+- An entity creation: `previous_data` field is `null` because there weren't previous data.
+- An entity deletion: `current_data` field is `null` because there is not new data.
+
+Each operation type have a structure that depends on its involved data.
+
+The structured data will be stored in JSON format because it's simpler to unmarshall it for sending
+it through the API to the clients.
+
+The update operation is needed because when an operation is caused by another one, we have to
```
[the remark this answers is not in the document]
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md]: I rephrased and added an example, please check again.

File: satellite/admin-back-office-v1.md
Comment (dlamarmorgan) [reply satellite/admin-back-office-v1.md:1]: Looks great now thanks!

File: satellite/admin-back-office-v1.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:255]: Unfortunately, we cannot add the rate limits to the account level and I had to remove the number of buckets limit too because we don't have those in the current satellite version.

If in the future, we add them, then we can add them to the admin back-office.

File: satellite/admin-back-office-v1.md
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:1]: After consulting with several mates, my understanding is that SOC2 compliance nodes are going to be categorized as being a specific placement, so the admin UI can show the list of values that an operator can set, and one of them is SOC2.

We don't need anything specific for handling SOC2, despite it being specified using node tags because indicating that a bucket only stores objects in SOC2 nodes will be specified in the same field as a normal geo fence https://github.com/storj/storj/blob/main/satellite/satellitedb/dbx/project_bucket.dbx#L65.

The user model has a field for placement, however, they are the default setting to apply to new projects. The same happens with the project model, but the default setting applies to new buckets. In the end, who really specifies if the data has to be placed in a specific area or compliance is the bucket.

In the future, we may have node tags that may map to different account, project, and bucket settings, however, they are not defined now. When we define them, we'll have to apply changes to the satellite back office as we do in the other areas of the satellite.

Despite all of these, I replaced geo fence by geo fence and SOC2 to make it clearer that we are considering SOC2 because it's something that will land in production before the back office implementation is completed.

**Review by egonelbre — APPROVED** [review]

File: satellite/admin-back-office-v1.md [hunk 14]
```diff
@@ -0,0 +1,646 @@
...
+- Account ID. What account was modified.
+- Entity that was modified (i.e. account, project, bucket) and ID.
+- Operation. What was done (e.g. change email address, set user agent, etc.). The text used to
+  identify the operation has to be unique inside of the entity and human friendly.
+- Current data. The new data that was modified with this operation. This is a structured blob
+  because it could be more than one modified field per operation.
+- Previous data. The existing data previous to the modification performed by this operation. This is
+  a structured blob because it could be more than one modified field per operation.
+
+This information is stored in the same database that the satellite uses.
```
Comment (littleskunk) [inline satellite/admin-back-office-v1.md:321]: The big question is why do we keep a record in the first place. Do we want to have a full audit log in case someone abuses the admin interface? If so we should not store the log in a location that the same person can most likely modify.

Let's say I want to leave the company and blow up everything on my way out. I would start to modify a bunch of users, projects and buckets and for each modification I make sure to delete the evidence from the database. At that point the log is kind of useless to catch me unless you protect the log in a way that I don't have permission to alter it.

Comment (mobyvb) [reply satellite/admin-back-office-v1.md:321]: I think that is a valid point, and it would be true in an extreme case. Still, I think having an audit log is useful regardless:
* perhaps someone who wants to blow up something will be deterred just by the fact of there being logs
* perhaps someone who wants to blow up something does not have `delete` abilities on the audit table

But personally, I think it is a great idea for one reason, completely not related to breaking things/malicious employees:

It gives us a very easy way to see if anything could have been caused by an admin if diagnosing issues for specific users. Let's say someone submits a support ticket due to an issue, or a customer complains about something to the sales team. Whoever is investigating would probably be interested in any recent admin actions on that customer, and why they were made. If there is a table, we can identify these actions as specific events that occurred at a point in time, which (imo) helps a lot for investigation work.

File: satellite/admin-back-office-v1.md [hunk 15]
```diff
@@ -0,0 +1,642 @@
...
+### Context
+
+Satellite currently has an [administration
+API](https://github.com/storj/storj/blob/main/satellite/admin/README.md)
+and minimalist [Web interface](https://github.com/storj/storj/tree/main/satellite/admin/ui) which
+allows to interact with its endpoints through Web forms.
+
+API authentication is conducted by two mechanism:
+- [An unique token secret](https://github.com/storj/storj/blob/main/satellite/admin/server.go#L215).
+  The service isn't exposed publicly, the only way to access to it is forwarding the port of the
```
[the remark this answers is not in the document]
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:35]: I didn't know that. First news.

File: satellite/admin-back-office-v1.md (hunk 14 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:321]: The modification history is more for what Moby said.
Moreover, we are showing it in the same web UI.

File: satellite/admin-back-office-v1.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:35]: Well, no https://github.com/storj/storj/blob/main/satellite/admin/server.go#L234

It seems that it changed at some point and if you log through Oauth2 proxy, the token secret isn't considered at at all.

File: satellite/admin-back-office-v1.md (hunk 14 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:321]: About "showing the history in the Web UI" you can see in the already-built frontend https://storj.github.io/admin-ui  For example, navigate to see a user details page and at the bottom of the page

image

File: satellite/admin-back-office-v1.md [hunk 16]
```diff
@@ -0,0 +1,646 @@
...
+- We use the term "operator" to refer to people who has access to the back office.
+
+## Essentials
+
+### Header
+
+Date: 2023-07-27
+Owner: Ivan Fraixedes
+
+Accountable: TBD
```
Comment (jenlij) [inline satellite/admin-back-office-v1.md:15]: I would say Team Delivery console squad (Jen)

**Review by jenlij — APPROVED** [review]
thorough and comprehensive, thank you!

File: satellite/admin-back-office-v1.md (hunk 16 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md]: Added

File: satellite/admin-back-office-v1.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (littleskunk) [reply satellite/admin-back-office-v1.md:36]: Ok I didn't try it since that commit. Good to know. Thanks.

File: satellite/admin-back-office-v1.md (hunk 14 above)
Comment (littleskunk) [reply satellite/admin-back-office-v1.md:322]: I think we can figure out that detail later. For now good to merge from my side. Adding a text field is nothing that is worth blocking the PR.

**Review by littleskunk — APPROVED** [review]

File: satellite/admin-back-office-v1.md (hunk 14 above)
Comment (ifraixedes) [reply satellite/admin-back-office-v1.md:322]: That's a good idea.
@mobyvb  thought something similar, but more generic https://github.com/storj/design-docs/pull/1#discussion_r1296310794

We could add what Moby said, and then the UI could lead the operator to set specific information when performing an action. I know that people could cheat by making an API call and bypassing UI, but this is for internal use, and I like to trust who is working at Storj, considering that some employee does it, we could find who did it and take whatever measures.

Anyway, as I commented to Moby, that's something that we can add later without that much hassle, and if we want it for this first initial version then:
a) If we add it without specifying what, it seems to add "premature features".
b) If we specify it now, we will delay the design doc for a while more, because we should have some meetings with product owners and Tome, to modify the use cases and the UI. 

Doing a) is bad, for the reasons that I gave to Moby.
Doing b) isn't worth it, for something that we can add without that much trouble in some of the next versions.

Nonetheless, I'm going to add this to the out-of-scope section as a reminder of this conversation and a feature that we may want/need.
